flocxclient/common/base.py

```diff
@@ -0,0 +1,103 @@
+"""
+Base utilities to build API operation managers and objects on top of.
+"""
+
+import logging
+import abc
+import six
+
+from flocxclient.common.apiclient import base
+
+
+LOG = logging.getLogger(__name__)
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Manager(object):
+    """Provides  CRUD operations with a particular API."""
+
+    @property
+    @abc.abstractmethod
+    def resource_class(self):
+        """The resource class
+        """
+
+    @property
+    @abc.abstractmethod
+    def _resource_name(self):
+        """The resource name.
+        """
+
+    def __init__(self, api):
+        self.api = api
+
+    def _path(self, resource_id=None):
+        """Returns a request path for a given resource identifier.
+        :param resource_id: Identifier of the resource to generate the request
+                            path.
+        """
+
+        # NOTE: currently the flocx-market service does not use versioned urls
+        # The below snippet would be correct if flocx-market used proper urls
```

Comment:
It's probably better to say "versioned urls" instead of "proper urls". We don't want to be tooooooo critical of our own code :)